Keycloak/config.sh

```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+
+# Get the database url from the environment variable
+url=$DATABASE_URL
+
+# Remove the not required content
+url=$(echo "$url" | sed 's/?sslmode=require//')
+
+# Extracting components from the database URL
+# This is done because Keycloak isn't allowing using the full database URL directly
+username=$(echo "$url" | awk -F '://' '{split($2,a,"@"); print a[1]}' | awk -F ':' '{print $1}')
+password=$(echo "$url" | awk -F '://' '{split($2,a,"@"); print a[1]}' | awk -F ':' '{print $2}')
+host=$(echo "$url" | awk -F '://' '{split($2,a,"@"); print a[2]}' | awk -F '?' '{print $1}')
+host_port_path=$(echo "$url" | awk -F '://' '{split($2,a,"@"); print a[2]}' | awk -F '?' '{print $1}')
+IFS='/' read -r host_port path <<< "$host_port_path"
+IFS=':' read -r host port <<< "$host_port"
+
+# Constructing the JDBC URL
+db_url="jdbc:postgresql://$host:$port/$path"
+
+# Remove the previous configuration
+rm -rf ./keycloak-$(cat ./KEYCLOAK_VERSION)/conf/keycloak.conf
+
+# Writing a new, updated configuration
+echo "db=postgres" > ./keycloak-$(cat ./KEYCLOAK_VERSION)/conf/keycloak.conf
```

Comment:
Let's not write the username, password and database URL to a file as people might accidentally commit these secrets then but instead export them as environment variable with `export DB_USERNAME="$username"`. 

We can then reference these in the ci.yml run stage like so:
--db-username=$DB_USERNAME

Reply:
Did you see my suggestions in Discord as to how the env vars could be passed in here?